### ISSUE TITLE
zodToJsonSchema() - Can't get the correct describe of optional() schema

### ISSUE THREAD

**Issue by getandplay** [StefanTerdell/zod-to-json-schema#66]
The zodToJsonSchema works fine to get the correct describe text if the property is not optional.

```typescript
const TestSchema = z.object({ title: z.string().min(1) });
const TheObjectSchema = z.object({
  p1: TestSchema.describe('aaaaaaaaa'),
  p2: TestSchema.describe('bbbbbbbbb'),
  p3: TestSchema.describe('ccccccccc'),
})
  .describe('sssssssss');

const schema = zodToJsonSchema(TheObjectSchema, {
  target: 'openApi3',
  $refStrategy: 'none',
});
console.log((schema as any).properties);
```
After running this code you can get the result correctly.
```
{
  p1: {
    type: 'object',
    properties: { title: [Object] },
    required: [ 'title' ],
    additionalProperties: false,
    description: 'aaaaaaaaa'
  },
  p2: {
    type: 'object',
    properties: { title: [Object] },
    required: [ 'title' ],
    additionalProperties: false,
    description: 'bbbbbbbbb'
  },
  p3: {
    type: 'object',
    properties: { title: [Object] },
    required: [ 'title' ],
    additionalProperties: false,
    description: 'ccccccccc'
  }
}
```
You can see that the descriptions are what we want.

But if you change it to optional(), it will fail.

``` ts
const TestSchema = z.object({ title: z.string().min(1) });
const TheObjectSchema = z.object({
  p1: TestSchema.optional().describe('aaaaaaaaa'),
  p2: TestSchema.optional().describe('bbbbbbbbb'),
  p3: TestSchema.optional().describe('ccccccccc'),
})
  .describe('sssssssss');

const schema = zodToJsonSchema(TheObjectSchema, {
  target: 'openApi3',
  $refStrategy: 'none',
});
console.log((schema as any).properties);
```
After running this code, you will get the following result.
```
{
  p1: {
    type: 'object',
    properties: { title: [Object] },
    required: [ 'title' ],
    additionalProperties: false,
    description: 'ccccccccc'
  },
  p2: {
    type: 'object',
    properties: { title: [Object] },
    required: [ 'title' ],
    additionalProperties: false,
    description: 'ccccccccc'
  },
  p3: {
    type: 'object',
    properties: { title: [Object] },
    required: [ 'title' ],
    additionalProperties: false,
    description: 'ccccccccc'
  }
}
```
You can see, all `descriptions` are the same, seems like it used the latest description, which is 'ccccccccc'.

Can I get someone to fix that? Or do you have any idea to avoid this error? 

I am using the latest version (3.21.1).

**Comment by getandplay** [comment]
BTW, if remove `$refStrategy: 'none',`, it can get the correct description. But the result is not I want.

**Comment by StefanTerdell** [comment]
Fix released in 3.21.2
